mixer/pkg/config/crd/admit.go

```diff
@@ -0,0 +1,387 @@
+// its name or location.
+func getAPIServerExtensionCACert(cl kubernetes.Interface) ([]byte, error) {
+	const name = "extension-apiserver-authentication"
+	c, err := cl.CoreV1().ConfigMaps(metav1.NamespaceSystem).Get(name, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	pem, ok := c.Data["requestheader-client-ca-file"]
+	if !ok {
+		return nil, fmt.Errorf("cannot find ca.crt in %v: ConfigMap.Data is %#v", name, c.Data)
+	}
+	return []byte(pem), nil
+}
+
+// MakeTLSConfig makes a TLS configuration suitable for use with the
+// GenericAdmissionWebhook.
+func makeTLSConfig(serverCert, serverKey, caCert []byte) (*tls.Config, error) {
+	caCertPool := x509.NewCertPool()
+	caCertPool.AppendCertsFromPEM(caCert)
+	cert, err := tls.X509KeyPair(serverCert, serverKey)
+	if err != nil {
+		return nil, err
+	}
+	return &tls.Config{
+		Certificates: []tls.Certificate{cert},
+		ClientCAs:    caCertPool,
+		ClientAuth:   tls.RequireAndVerifyClientCert,
+	}, nil
+}
+
+func getKeyCertsFromSecret(client kubernetes.Interface, name, namespace string) (serverKey, serverCert, caCert []byte, err error) { // nolint: lll
+	listWatch := cache.NewListWatchFromClient(client.CoreV1().RESTClient(),
+		"secrets", namespace, fields.OneTermEqualSelector("metadata.name", name))
+	var secret *v1.Secret
+	stop := make(chan struct{})
+	_, controller := cache.NewInformer(listWatch, &v1.Secret{}, 30*time.Second,
+		cache.ResourceEventHandlerFuncs{
+			AddFunc: func(obj interface{}) {
+				if secret == nil {
+					secret = obj.(*v1.Secret)
+					close(stop)
+				}
+			},
+		},
+	)
+	controller.Run(stop)
+
+	var ok bool
+	if serverKey, ok = secret.Data[secretServerKey]; !ok {
+		return nil, nil, nil, errors.New("server key missing")
+	}
+	if serverCert, ok = secret.Data[secretServerCert]; !ok {
+		return nil, nil, nil, errors.New("server cert missing")
+	}
+	if caCert, ok = secret.Data[secretCACert]; !ok {
+		return nil, nil, nil, errors.New("ca cert missing")
+	}
+	return serverKey, serverCert, caCert, nil
+}
+
+// NewController creates a new instance of the admission webhook controller.
+func NewController(client kubernetes.Interface, options ControllerOptions) (*AdmissionController, error) {
+	return &AdmissionController{
+		client:  client,
+		options: options,
+	}, nil
+}
+
+func setup(client kubernetes.Interface, options *ControllerOptions) (*tls.Config, []byte, error) {
+	apiServerCACert, err := getAPIServerExtensionCACert(client)
+	if err != nil {
+		return nil, nil, err
+	}
+	serverKey, serverCert, caCert, err := getKeyCertsFromSecret(
+		client, options.SecretName, options.ServiceNamespace)
+	if err != nil {
+		return nil, nil, err
+	}
+	tlsConfig, err := makeTLSConfig(serverCert, serverKey, apiServerCACert)
+	if err != nil {
+		return nil, nil, err
+	}
+	return tlsConfig, caCert, nil
+}
+
+// Run implements the admission controller run loop.
+func (ac *AdmissionController) Run(stop <-chan struct{}) {
```

Comment:
The referenced issue has been fixed. https://github.com/kubernetes/kubernetes/pull/50476
Either fix it now, or file an issue jointly with pilot to address it afterwards.

Reply:
It's not good to fix here locally, better to be aligned with pilot webhooks. Filed https://github.com/istio/istio/issues/1795, and change the TODO reference to the new issue for the tracking. I'll keep my eyes on it.